stdlib/public/SwiftShims/GlobalObjects.h

```diff
@@ -102,13 +102,6 @@ static_assert(
     4 * sizeof(__swift_intptr_t) + sizeof(__swift_int64_t),
   "_SwiftSetBodyStorage has unexpected size");
 
-static_assert(std::is_pod<_SwiftEmptyArrayStorage>::value,
-              "empty array type should be POD");
-static_assert(std::is_pod<_SwiftEmptyDictionarySingleton>::value,
-              "empty dictionary type should be POD");
-static_assert(std::is_pod<_SwiftEmptySetSingleton>::value,
-              "empty set type should be POD");
-
```

Comment:
IIUC, the issue is caused by `HeapObject`, so we could (and probably should) still check for the PODness of `_Swift{Array,Dictionary,Set}BodyStorage` here.

Reply:
Yes, the issue was caused by `HeapObject`.  However, the singletons contain a `HeapObject` type, and that fails the PODness check  :-(.

Reply:
Sure, but the `*BodyStorage` types are still PODs. (Checking those is of limited, but probably nonzero, benefit.)

Reply:
@lorentey - hmm, interesting, since I definitely am seeing those checks fail as well.  I can try to dig through those and split this up into two sets, one to remove the other assertions, and one to deal with the PODness here.

Reply:
Oh, that's weird! They're simple structs of integers and pointers with no explicit constructors/destructors. Are typedefs like `__swift_intptr_t` not considered POD on MSVC?

The stdlib's Swift code also assumes these are trivial types -- for example, it doesn't distinguish between initialization and assignment when creating such objects.

Reply:
I'm almost 100% certain that it is what I said earlier:

```
struct _SwiftEmptyDictionarySingleton {
  struct HeapObject header;
  struct _SwiftDictionaryBodyStorage body;
  __swift_uintptr_t metadata;
};
```

Note that the structure itself contains an **inline** copy of `HeapObject`.  That breaks the PODness of the type, because the `HeapObject` is no longer a POD type due to the `std::atomic<uint64_t>` in it breaking that.

Reply:
I'm suggesting to check `*BodyStorage` for podness, not `*Singleton`.

Reply:
Bleh, apparently, I can't read :-(.  The removed assertions here are for the Singletons, and the `_SwiftEmptyArrayStorage` which has an inline `HeapObject` header.